pkg/apis/eventsource/v1alpha1/types.go

```diff
@@ -865,16 +865,21 @@ type BitbucketServerEventSource struct {
 	// Webhook holds configuration to run a http server
 	Webhook *WebhookContext `json:"webhook,omitempty" protobuf:"bytes,1,opt,name=webhook"`
 	// ProjectKey is the key of project for which integration needs to setup
-	ProjectKey string `json:"projectKey" protobuf:"bytes,2,opt,name=projectKey"`
+	// +optional
+	ProjectKey string `json:"projectKey,omitempty" protobuf:"bytes,2,opt,name=projectKey"`
```

Comment:
Are we deprecating this? If yes, it would be better to mention it in the comments, also give a version when it will be unsupported. For this one, we can stop supporting it in v1.8.

Reply:
Yes I was wondering if we should deprecate it - v1.8 sounds good to me! Will add the comments right now